apps/leave-app/webapp/src/view/GeneralLeave/component/LeaveSelection.tsx

```diff
@@ -20,26 +20,44 @@ import PregnantWomanIcon from "@mui/icons-material/PregnantWoman";
 import WorkOffIcon from "@mui/icons-material/WorkOff";
 import { Stack, Typography, useTheme } from "@mui/material";
 
-import { useState } from "react";
-
 import DatePill from "./DatePill";
 import LeaveSelectionIcon from "./LeaveSelectionIcon";
 
-export default function LeaveSelection() {
+interface LeaveSelectionProps {
+  daysSelected: number;
+  selectedLeaveType: string;
+  onLeaveTypeChange: (leaveType: string) => void;
```

Comment:
[IF RELEVANT] If leave types are previously defined (I saw "full" "half" and something else defined above) use that type here for type safety
```suggestion
  onLeaveTypeChange: (leaveType: "full" | "half" | "other type") => void;
```